### ISSUE TITLE
Allow stack config to be written in language of Pulumi program

### ISSUE THREAD

**Issue by pierskarsenbarg** [pulumi/pulumi#14279]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Currently, stack config is only in YAML. Having it in the same language as the Pulumi project would make it more readable when it gets big. 

You can, of course, not use the stack config at all, but it's necessary for default provider config as well as secrets.

**Comment by Frassle** [comment]
Configuring default providers is covered by https://github.com/pulumi/pulumi/issues/2059.

The secrets one is interesting. My first thought was that if we had a CLI command to encrypt a value and a library function to decrypt it you could use that, but then I realised that secret encryption/decryption is _per stack_ but program source code is _per project_, so I think we'd need https://github.com/pulumi/pulumi/issues/11549 done first.

**Comment by pierskarsenbarg** [comment]
@Frassle program source code is per project, but when it's run it's per stack, right? 

![](https://media.giphy.com/media/l0HlIbzTDGWY0ySly/giphy.gif)

**Comment by Frassle** [comment]
> program source code is per project, but when it's run it's per stack, right?

Yes, but if the config is written in the program source code then you either need some map of stack name to config item in the source code, or you need to treat it as per-project.

Hard to see what per-stack in language config would look like?
